packages/client/src/features/game/Game.tsx

```diff
@@ -0,0 +1,470 @@
+import { useEffect, useRef, useState } from 'react'
+import styles from './styles.module.css'
+import { TBattleCart, TCard } from './types'
+import { useWindowSize } from '../../shared/hooks/useWindowSize'
+import imports from './imports'
+import { button_text, notice_game } from './assets'
+
+export const Game = () => {
+  const [widthGame, setWidthGame] = useState(0)
+  const [heightGame, setHeightGame] = useState(0)
+  const [isStartGame, setStartGame] = useState(false)
+  const [deckCards, setDeckCards] = useState<TCard[]>([])
+  const [trumpCard, setTrumpCard] = useState<TCard | null>(null)
+
+  const [isFirstGetCards, setFirstGetCards] = useState(false)
+  const [playerCards, setPlayerCards] = useState<TCard[]>([])
+  const [botCards, setBotCards] = useState<TCard[]>([])
+
+  const [isNoticeText, setNoticeText] = useState('')
+
+  const canvasRef = useRef<HTMLCanvasElement>(null)
+  const canvas = canvasRef.current
+  const ctx = canvas?.getContext('2d')
+  const [width, height] = useWindowSize()
+
+  const [selectedSrcCardToMove, setSelectedSrcCardToMove] = useState('')
+  const [isMoveBot, setMoveBot] = useState(false)
+  const [isMovePlayer, setMovePlayer] = useState(false)
+
+  const [battleCards, setBattleCards] = useState<TBattleCart[]>([])
+  const [leftCards, setLeftCards] = useState<TBattleCart[]>([])
+  const [buttonText, setButtonText] = useState('')
+  const [isPlayer, setPlayer] = useState<boolean | undefined>(undefined)
+
+  const endGame =
+    (playerCards.length === 0 || botCards.length === 0) &&
+    deckCards.length === 0
+
+  useEffect(() => {
+    setWidthGame(Math.round(width - (width * 5 * 2) / 100))
+    setHeightGame(Math.round(height - imports.LOGO_HEIGHT - 30))
+  }, [width, height])
+
+  useEffect(() => {
+    if (ctx) {
+      ctx.clearRect(0, 0, widthGame, widthGame)
+    }
+    if (!trumpCard) {
+      setStartGame(true)
+    }
+  }, [widthGame, heightGame])
+
+  useEffect(() => {
+    if (isStartGame) {
+      const arr = imports.shuffle(imports.initialDeckCard)
+      setTrumpCard(arr[0])
+      setDeckCards(arr)
+      setFirstGetCards(true)
+    }
+  }, [isStartGame])
+
+  useEffect(() => {
+    if (isFirstGetCards) {
+      setPlayerCards(deckCards.slice(30, 36))
+      setBotCards(deckCards.slice(24, 30))
+    }
+  }, [isFirstGetCards])
+
+  useEffect(() => {
+    if (playerCards.length !== 0 && botCards.length !== 0 && isFirstGetCards) {
```

Comment:
_Не критично, но лучше исправить_

Конкретно тут, но и в многих местах дальше вижу большой уровень вложенности if'ов. Всегда по возможности следует избегать большой вложенности, тк это вызывает сложности восприятия кода. 

Не заставляю переписывать все такие места, тк в движке игры, порой, не обойтись без вложенности, но как минимум можно убрать 1 слой, переписав логику первого `if` на обратную и выходя при этом из логики. Предлагаю пройтись по диффу и поправить все такие места.

```suggestion
    if (!(playerCards.length !== 0 && botCards.length !== 0 && isFirstGetCards)) return;
```